python/pyarrow/tests/test_convert_pandas.py

```diff
@@ -160,9 +160,35 @@ def test_integer_index_column(self):
         df = pd.DataFrame([(1, 'a'), (2, 'b'), (3, 'c')])
         _check_pandas_roundtrip(df, preserve_index=True)
 
+    def test_index_metadata_field_name(self):
+        df = pd.DataFrame(
+            [(1, 'a', 3.1), (2, 'b', 2.2), (3, 'c', 1.3)],
+            index=pd.MultiIndex.from_arrays(
+                [['c', 'b', 'a'], [3, 2, 1]],
+                names=[None, 'foo']
+            )
+        ).rename(columns=dict(zip(range(3), ['a', None, 'c'])))
+        t = pa.Table.from_pandas(df, preserve_index=True)
+        raw_metadata = t.schema.metadata
+
+        js = json.loads(raw_metadata[b'pandas'].decode('utf8'))
+
+        col1, col2, col3, idx0, foo = js['columns']
+
+        assert col1['name'] == col1['field_name']
+        assert col2['name'] is None
+        assert col2['field_name'] is None
```

Comment:
So in principle, IMO this should be the string `"None"` to make `field_name` perfectly usable for schema lookup. 
Otherwise you need to do `schema.get_field_index(field_name or "None")`

Reply:
Hm, this works without that, let me see what's happening.

Reply:
We're actually explicitly handling this case later on in `table_to_blockmanager` so this is okay.

Reply:
Yes, the current code works without (just as it did work before with a "name" of None) as it is handled by `table_to_blockmanager`. 
But that means that you will always have to special case this option, and for me that should be the point of "field_name" that ``schema.get_field_index(field_name)`` is guaranteed to not error (and you thus don't have to special case None)

Reply:
Ok, I've implemented this. Pushing it up now.